frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -3,11 +3,8 @@ This module provides a way to display book information (filename and book metada
 ]]
 
 local BD = require("ui/bidi")
-local DocSettings = require("docsettings")
 local DocumentRegistry = require("document/documentregistry")
-local ImageViewer = require("ui/widget/imageviewer")
 local InfoMessage = require("ui/widget/infomessage")
-local KeyValuePage = require("ui/widget/keyvaluepage")
```

Comment:
Why get rid of these at the top, to go use the uglier `local widget = require("ui/widget/keyvaluepage"):new{...}` below ?
I don't think there's any risk of require loop here.
I admit I sometime do it when it's a rarely used module, to wait to load it - so it might be fine for ImageViewer and KeyBaluePage - but DocSettings is probably always loaded, so it doesn't really need this treatment.

Reply:
The same for KeyValuePage: if we ask to show book description only, it is not needed.

Reply:
I understand you do not like `require():new` and prefer `local` before?

Reply:
Yes, but then, why not do the same for InfoMessage and require it in onShowBookCover() and onShowBookDescription(), which are both rarely used?

It's also not the number of times it is used that should make it be at the top, rather how frequent/heavy it is, and if other modules would already have require() it and it is most probably already in memory. ie. for KeyValuePage, readerdictionary.lua does it at the top, so it's already in memory (or soon will be in your "start in file browser").

If you start with doing that, we'll shrink the top of our files quite heavily :)
And `require("ui/widget/keyvaluepage"):new{...}` is just ugly. It's fine for quick shorthands like needing a function form util.lua, but otherwise, it brings nothing.

Reply:
> I understand you do not like require():new and prefer local before?

Yes, I would prefer a local = require(...) on one line and the :new{} on the next, it's more readable - and what we mostly use.
(But the little `require("util").doSomeSstuff()` are small and fine).

Just saying, do not go too crazy with moving them all into where the function they are used. It feels OK for ImaveViewer, which we seldom need, also OK for KeyValuePage even if we do require it at top elsewhere, but DocSettings is so common.